ted2zim/scraper.py

```diff
@@ -199,53 +217,58 @@ def extract_video_info(self, url):
         keywords = self.soup.find("meta", attrs={"name": "keywords"})["content"]
         keywords = [key.strip() for key in keywords.split(",")]
 
-        # Extract the ratings list for the TED talk
-        # ratings = talk_info["ratings"]
-
-        # Append the meta-data to a list
-        self.videos.append(
-            [
-                {
-                    "id": video_id,
-                    "title": title.encode("utf-8", "ignore"),
-                    "description": description.encode("utf-8", "ignore"),
-                    "speaker": speaker.encode("utf-8", "ignore"),
-                    "speaker_profession": speaker_profession.encode("utf-8", "ignore"),
-                    "speaker_bio": speaker_bio.encode("utf-8", "ignore"),
-                    "speaker_picture": speaker_picture.encode("utf-8", "ignore"),
-                    "date": date.encode("utf-8", "ignore"),
-                    "thumbnail": thumbnail.encode("utf-8", "ignore"),
-                    "video_link": video_link.encode("utf-8", "ignore"),
-                    "length": length,
-                    "subtitles": subtitles,
-                    "keywords": keywords,
-                    "ratings": ratings,
-                }
-            ]
-        )
+        # Check if video ID already exists. If not, append data to self.videos
+        if not any(video[0].get("id", None) == video_id for video in self.videos):
+            self.videos.append(
+                [
+                    {
+                        "id": video_id,
+                        "title": title,
+                        "description": description,
+                        "speaker": speaker,
+                        "speaker_profession": speaker_profession,
+                        "speaker_bio": speaker_bio,
+                        "speaker_picture": speaker_picture,
+                        "date": date,
+                        "thumbnail": thumbnail,
+                        "video_link": video_link,
+                        "length": length,
+                        "subtitles": subtitles,
+                        "keywords": keywords,
+                    }
+                ]
+            )
+            logger.debug(f"Successfully inserted video {video_id} into video list")
+        else:
+            logger.debug(f"Video {video_id} already present in video list")
 
     def dump_data(self):
-        """
-        Dump all the data about every TED talk in a json file
-        inside the 'build' folder.
-        """
-        # Prettified json dump
-        data = json.dumps(self.videos, indent=4, separators=(",", ": "))
+
+        # Dump all the data about every TED talk in a json file
+        # inside the 'build' folder.
+        logger.debug(
+            f"Dumping {len(self.videos)} videos into {self.ted_videos_json} and {len(self.categories)} categories into {self.ted_categories_json}"
+        )
+        video_data = json.dumps(self.videos, indent=4, separators=(",", ": "))
```

Comment:
remove the separators arg. those are the defaults in py3.

Reply:
Done